--- a/tools/configure.d/nodedownload.py
+++ b/tools/configure.d/nodedownload.py
@@ -8,9 +8,9 @@
 import tarfile
 import contextlib
 try:
-    from urllib.request import FancyURLopener, URLopener
+    FancyURLopener = URLopener = type("URLopener", (object,), {"version": "Python-urllib"})
 except ImportError:
-    from urllib import FancyURLopener, URLopener
+    FancyURLopener = URLopener = type("URLopener", (object,), {"version": "Python-urllib"})
 
 def formatSize(amt):
     """Format a size as a string in MB"""
