--- a/deps/uv/src/unix/process.c
+++ b/deps/uv/src/unix/process.c
@@ -407,7 +407,7 @@
 }
 
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && !defined(UV_NO_POSIX_SPAWN)
 typedef struct uv__posix_spawn_fncs_tag {
   struct {
     int (*addchdir_np)(const posix_spawn_file_actions_t *, const char *);
@@ -859,7 +859,7 @@
   int exec_errorno;
   ssize_t r;
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && !defined(UV_NO_POSIX_SPAWN)
   uv_once(&posix_spawn_init_once, uv__spawn_init_posix_spawn);
 
   /* Special child process spawn case for macOS Big Sur (11.0) onwards
